api/organisations/models.py

```diff
@@ -545,6 +553,28 @@ def _get_default_subscription_metadata_kwargs(self) -> dict[str, Any]:
             "feature_history_visibility_days": self.feature_history_visibility_days,
         }
 
+    def is_billing_terms_dates_set(self) -> bool:
+        return (
+            self.current_billing_term_starts_at is not None
+            and self.current_billing_term_ends_at is not None
+        )
+
+    def has_active_billing_periods(self) -> bool:
+        """
+        Returns True if current date is within the billing term.
+        If either start or end date is None, returns False.
+        """
+        if not self.is_billing_terms_dates_set():
+            return False
+        assert self.current_billing_term_starts_at is not None
+        assert self.current_billing_term_ends_at is not None
```

Comment:
I understand this is needed for mypy to pass, so let's make it self-evident:

```suggestion
    if TYPE_CHECKING:
        assert self.current_billing_term_starts_at is not None
        assert self.current_billing_term_ends_at is not None
```